collection.go

```diff
@@ -452,27 +450,21 @@ func (cl *collectionLoader) loadProgram(progName string) (*Program, error) {
 	for i := range progSpec.Instructions {
 		ins := &progSpec.Instructions[i]
 
-		if !ins.IsLoadFromMap() || ins.Reference == "" {
-			continue
-		}
-
-		if uint32(ins.Constant) != math.MaxUint32 {
```

Comment:
I think this might incur an unnecessary breaking change since we've never enforced clearing the Reference field if the caller rewrites their own map fds.

@lmb any thoughts?

Reply:
iirc, I removed the check at some point because it was causing issues with the tests, but adding it back now doesn't break the tests. So I guess keeping it doesn't hurt, and allows people to rewrite their own FDs if they so wish.